### PR TITLE
python312Packages.materialyoucolor: init at 2.0.9

### LINKED ISSUE #318557: Package request: materialyoucolor-python
**Project description**
[Material You color algorithms](https://m3.material.io/styles/color/overview) for python!

It is built in reference with offical [typescript implementation](https://github.com/material-foundation/material-color-utilities/tree/main/typescript) except it's color quantization part, which is based on [c++ implementation](https://github.com/material-foundation/material-color-utilities/tree/main/cpp) thanks to [pybind](https://github.com/pybind).

**Metadata**

* homepage URL: https://github.com/T-Dynamos/materialyoucolor-python
* source URL: https://github.com/T-Dynamos/materialyoucolor-python
* license: mit
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/318557

This PR adds the `materialyoucolor` python package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Moraxyc** [comment]
Result of `nixpkgs-review pr 318929` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.materialyoucolor</li>
    <li>python311Packages.materialyoucolor.dist</li>
    <li>python312Packages.materialyoucolor</li>
    <li>python312Packages.materialyoucolor.dist</li>
  </ul>
</details>

**Review by Moraxyc — APPROVED** [review]
LGTM, thanks

**Comment by lVentus** [comment]
when can we merge it?

**Comment by lVentus** [comment]
Hello? Anyone? @wegank

File: pkgs/development/python-modules/materialyoucolor/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  rich,
+  pillow,
+}:
+
+let
+  test-image = fetchurl {
+    name = "test-image.jpg";
+    url = "https://unsplash.com/photos/u9tAl8WR3DI/download";
+    hash = "sha256-shGNdgOOydgGBtl/JCbTJ0AYgl+2xWvCgHBL+bEoTaE=";
+  };
```
Comment (wegank) [inline pkgs/development/python-modules/materialyoucolor/default.nix:18]: Is this test image free? If not, I would take it out of `checkPhase` and probably put it in `passthru.tests` independent of the derivation.

Comment (lVentus) [reply pkgs/development/python-modules/materialyoucolor/default.nix:18]: im not sure. I am the one who requested this package, not the one who wrote the nix file. Sorry

Comment (WoLeo-Z) [reply pkgs/development/python-modules/materialyoucolor/default.nix:18]: https://unsplash.com/photos/low-angle-photography-coconut-trees-during-golden-hour-u9tAl8WR3DI
Free to use under the [Unsplash License](https://unsplash.com/license) .

**Review by WoLeo-Z — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 318929` at 4b19fe22 run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python311Packages.materialyoucolor</li>
    <li>python312Packages.materialyoucolor</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**